### ISSUE TITLE
Exclude /.idea/workspace.xml

### ISSUE THREAD

**Issue by ivoba** [withastro/astro#4325]
### What version of `astro` are you using?

1.0.5

### Are you using an SSR adapter? If so, which one?

no

### What package manager are you using?

npm

### What operating system are you using?

Linux

### Describe the Bug

When working with IntelliJ Ultimate the watcher keeps updating workspace.xml.

> 09:20:00 [astro] update /.idea/workspace.xml

Can this be excluded as this is a metafile of IntelliJ and not relevant for the project?

### Link to Minimal Reproducible Example

https://stackblitz/none

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by matthewp** [comment]
Can't you just add this file to your `.gitignore` file?

**Comment by ivoba** [comment]
@matthewp what i mean was that astro watches the `.idea/workspace.xml` file.
Not that it edits or updates the file. That was maybe misleading.

I would like to exclude this file from the watcher since it is not relevant for rerendering or hotloading.

**Comment by matthewp** [comment]
@ivoba got it, does the fact that it is watching this file cause any sort of problems?

**Comment by ivoba** [comment]
@matthewp not really. It just might cause some unnecessary rebuilds.
The display just confused me since it is not a source file per se.

**Comment by matthewp** [comment]
Maybe we should not be watching files that start with `.`. Pinging @bluwy as he might have encountered this request in Vite before.

**Comment by bluwy** [comment]
We should still keep watching files that start with `.`, e.g. SvelteKit that generates files under `.svelte-kit` that are imported in user code. I think the better way, since this is a nice-to-have, is to configure files to exclude watching through [`vite.server.watch.ignored`](https://vitejs.dev/config/server-options.html#server-watch).

**Comment by matthewp** [comment]
Ah, good point. @ivoba you can configure this in your astro.config.mjs like this:

```js
import { defineConfig } from 'astro/config';

export default defineConfig({
  vite: {
    server: {
      watch: {
        ignored: ['.idea/workspace.xml']
      }
    }
  }
}
```

Going to close for now, if enough people run into this being a problem for them we might add it as a default in the future.

**Comment by ivoba** [comment]
Thanks that was what i was looking for.
I added but however it seems it has no effect.

**Comment by Dragonlord1005** [comment]
> Thanks that was what i was looking for. I added but however it seems it has no effect.

`
import { defineConfig } from 'astro/config';

export default defineConfig({
  vite: {
    server: {
      watch: {
        ignored: ['**/.idea/workspace.xml']
      }
    }
  }
}
`
This should fix it

**Comment by rshurts** [comment]
You may as well cover everything in the .idea folder:

```js
	vite: {
		server: {
			watch: {
				ignored: ['**/.idea/**']
			}
		}
	}
```

**Comment by kaorukobo** [comment]
@rshurts This worked perfectly for me, thanks. ...while it is curious that just `ignored: ['.idea/**']` does not work.

**Comment by FDiskas** [comment]
I would say - everything that is inside the `.gitignore` file, should also be ignored.

**Comment by fwextensions** [comment]
It would be nice if this solution were more discoverable.  I had to try several different search queries to find this one issue.  Then I went back to the [config docs](https://docs.astro.build/en/reference/configuration-reference/) to see if I'd missed something, but `watch` doesn't appear on that page, and `ignore` only in the context of trailing slashes.  And you'd have to go several pages into the Vite docs to find the right option.

Also, for some reason, just adding a path like `"foo.md"` or "`./foo.md"` to the ignored list didn't do anything.  Saving that file would always cause a rebuild, even though the file's at the top level of the project.

For anyone else running into the same problem, this ended up working (no idea why it's necessary):

```
vite: {
  server: {
    watch: {
      ignored: [
        '**/.idea/**',
        '**/foo.md'
      ]
    }
  }
}
```
